app/sign.js

```diff
@@ -0,0 +1,265 @@
+
+  _.forEach(recoveryRequest.inputs, function(input, i) {
+    const isBech32 = !input.redeemScript;
+    const isSegwit = !!input.witnessScript;
+
+    // chain paths come from the SDK with a leading /, which is technically not allowed by BIP32
+    if (input.chainPath.startsWith('/')) {
+      input.chainPath = input.chainPath.slice(1);
+    }
+
+    const derivedHDNode = backupKeyNode.derivePath(input.chainPath);
+
+    console.log(`Signing input ${ i + 1 } of ${ recoveryRequest.inputs.length } with ${ derivedHDNode.neutered().toBase58() } (${ input.chainPath })`);
+
+    if (isBech32) {
+      const witnessScript = Buffer.from(input.witnessScript, 'hex');
+      const witnessScriptHash = utxoLib.crypto.sha256(witnessScript);
+      const prevOutScript = utxoLib.script.witnessScriptHash.output.encode(witnessScriptHash);
+      txBuilder.sign(i, derivedHDNode.keyPair, prevOutScript, utxoLib.Transaction.SIGHASH_ALL, input.amount, witnessScript);
+    } else {
+      const redeemScript = new Buffer(input.redeemScript, 'hex');
+
+      if (isSegwit) {
+        const witnessScript = new Buffer(input.witnessScript, 'hex');
+        txBuilder.sign(i, derivedHDNode.keyPair, redeemScript, utxoLib.Transaction.SIGHASH_ALL, input.amount, witnessScript)
+      } else {
+        txBuilder.sign(i, derivedHDNode.keyPair, redeemScript, utxoLib.Transaction.SIGHASH_ALL);
+      }
+    }
+  });
+
+  return txBuilder.build().toHex();
+};
+
+const handleSignEthereum = function(recoveryRequest, key, skipConfirm) {
+  const transaction = new EthTx(recoveryRequest.tx);
+
+  const customMessage = recoveryRequest.custom ? recoveryRequest.custom.message : 'None';
+  const txData = transaction.data;
+  const outputAddress = '0x' + txData.slice(16, 36).toString('hex');
+  const outputAmountWei = new BN(txData.slice(36, 68).toString('hex'), 16);
+  const outputAmount = outputAmountWei.div(WEI_PER_ETH).toString();
+
+  console.log('Sign Recovery Transaction');
+  console.log('=========================');
+  console.log(`Backup Key: ${ recoveryRequest.backupKey }`);
+  console.log(`Output Address: ${ outputAddress }`);
+  console.log(`Output Amount: ${ outputAmount }`);
+  console.log(`Custom Message: ${ customMessage }`);
+  console.log('=========================');
+
+  if (!key) {
+    console.log('Please enter the xprv of the wallet for signing: ');
+    key = prompt();
+  }
+
+  let backupKeyNode;
+
+  try {
+    backupKeyNode = utxoLib.HDNode.fromBase58(key);
+  } catch (e) {
+    throw new Error('invalid private key');
+  }
+
+  if (backupKeyNode.toBase58() === backupKeyNode.neutered().toBase58()) {
+    throw new Error('please provide the private (not public) wallet key');
+  }
+
+  if (backupKeyNode.neutered().toBase58() !== recoveryRequest.backupKey) {
+    throw new Error('provided private key does not match public key specified with recovery request');
+  }
+
+  if (!skipConfirm) {
```

Comment:
Throw this confirmy branch into a separate function, it's duped in a couple other places